Add Kitty support

trekdemo opened the pull request:
The strategy adds support for `:Make`/`:Dispatch` and `:Start`/`:Spawn` task execution using the [Kitty](https://sw.kovidgoyal.net/kitty) terminal emulator's windows and tabs.

`:Make`/`:Dispatch` tasks use kitty's get-text[^1] feature to avoid piping command outputs. This allows interactive debugging sessions and complex TUIs to function. It also allows outputs to display in color during execution and capture the output without ANSI codes.

Environment isolation is implemented using the `--copy-env` and `--env` flags. Environment variables are copied over, except the ones set by (neo)vim itself.

[^1]: https://sw.kovidgoyal.net/kitty/remote-control/#kitten-get-text

https://github.com/user-attachments/assets/4bf1b6b5-0067-4239-9cfa-03c70804fcc0

---

👋 Hey Mr. Pope,

A huge fan here. I've used this plugin for the last decade. I'm grateful for your work and glad that this plugin helped me have fun and earn a living.

I used this plugin with the tmux strategy since the beginning. Lately, I grew apart from tmux and started using the multiplexing capabilities of the Kitty terminal emulator. 

I've been testing this patch for the last month and have fixed all the issues I encountered. It's my first major Vimscript project, so I'm unsure if everything meets the standards.

Please let me know if this strategy is something you would like to add to this esteemed plugin.

trekdemo committed:
Implement the kitty strategy

The strategy adds support for :Make/:Dispatch and :Start/:Spawn
task execution using Kitty windows and tabs.

:Make/:Dispatch tasks use kitty's get-text[^1] feature to avoid piping
command outputs. This allows interactive debugging sessions and complex
TUIs to function. It also allows outputs to display in color during
execution and capture the output without ANSI codes.

Environment isolation is implemented using the `--copy-env` and `--env`
flags. Environment variables are copied over except the ones set by
(neo)vim itself.

[^1]: https://sw.kovidgoyal.net/kitty/remote-control/#kitten-get-text

trekdemo committed:
Add help section for kitty

kelvinst commented:
I have being using it and it works like a charm. Maybe just a good idea to be clearer on the instruction about setting remote control. I mean, I was just setting as `allow_remote_control yes` and was not setting `listen_on` option in `kitty.conf`. After I did that it work perfectly

trekdemo commented:
> I have being using it and it works like a charm. Maybe just a good idea to be clearer on the instruction about setting remote control. I mean, I was just setting as `allow_remote_control yes` and was not setting `listen_on` option in `kitty.conf`. After I did that it work perfectly

@kelvinst, thank you for the tip. I added an example kitty config with some explanation and a reference to the neovim issue tracker.

kelvinst commented on autoload/dispatch/kitty.vim:
```diff
@@ -0,0 +1,113 @@
+
+  let cmd = s:kitty_launch_cmd(type, a:request, dispatch#prepare_start(a:request, cmd_with_capturing, 'make'))
+  let output = system(cmd)
+  let window_id = matchstr(output, '^\d\+$')
```
@trekdemo, for some reason, this was not matching anymore after I updated kitty, maybe a trailing `\n`, IDK. Anyway, I think it's safe to just search for the first sequence of digits, no?

```suggestion
  let window_id = matchstr(output, '\d\+')
```

kelvinst replied:
This was causing dispatch to spawn duplicated jobs, because the kitty one was started, but it failed to get the window_id, so this returns exit code 1 and dispatch would then fallback to the other handlers (jobstart, terminal, etc). The symptom for me was that a quickfix was shown besides at the same time of the kitty window, which didn't bother me in the beginning, until I started getting some problem because of the concurrent processes trying to write to the same SQLite database.

trekdemo replied:
@kelvinst thank you for both documenting the bug and providing a fix as well.

I updated the PR/branch with your fix. 👍

trekdemo committed:
Fix Kitty window_id retrieval

The Kitty handler was failing to get the window_id and exiting with code
1,
causing dispatch to fallback to alternative handlers (jobstart,
terminal, etc).
This resulted in duplicate jobs.

In some environments additional (special) characters follow the window
ID.

Co-authored-by: Kelvin Stinghen <kelvin.stinghen@me.com>
